src/app/hooks/useDiscover.ts

```diff
@@ -0,0 +1,20 @@
+import useFetch from './useFetch';
+import { Movie } from '../../lib/types';
+
+export default function useDiscover(genre: string): {
+  movies: Movie[] | null;
+  isLoading: boolean;
+  errorMessage: string | null;
+} {
+  const {
+    data: movies,
+    isLoading,
+    errorMessage,
+  } = useFetch<Movie[]>(`/api/movie/discover?genre=${genre}`);
+
```

Comment:
I think a `useEffect` here would be useful if `genre` changes to fetch movies with different genres